src/full/Agda/TypeChecking/Monad/Constraints.hs

```diff
@@ -181,32 +192,31 @@ addAwakeConstraint' = addConstraintTo stAwakeConstraints
 
 addConstraintTo :: Lens' TCState Constraints -> Blocker -> Constraint -> TCM ()
 addConstraintTo bucket unblock c = do
-    pc <- build
+    pc <- buildConstraint unblock c
     stDirty `setTCLens` True
     bucket `modifyTCLens` (pc :)
-  where
-    build | isBlocking c = buildConstraint unblock c
-          | otherwise    = buildProblemConstraint_ unblock c
-    isBlocking = \case
-      SortCmp{}        -> False
-      LevelCmp{}       -> False
-      FindInstance{}   -> False
-      ResolveInstanceHead{} -> False
-      HasBiggerSort{}  -> False
-      HasPTSRule{}     -> False
-      CheckDataSort{}  -> False
-      ValueCmp{}       -> True
-      ValueCmpOnFace{} -> True
-      ElimCmp{}        -> True
-      UnBlock{}        -> True
-      IsEmpty{}        -> True
-      CheckSizeLtSat{} -> True
-      CheckFunDef{}    -> True
-      UnquoteTactic{}  -> True
-      CheckMetaInst{}  -> True
-      CheckType{}      -> True
-      CheckLockedVars{} -> True
-      UsableAtModality{} -> True
+
+isBlockingConstraint :: Constraint -> Bool
+isBlockingConstraint = \case
```

Comment:
Can we get documentation for in what sense these are blocking?

Reply:
I added a comment for what the basic idea is, but I don't think we have any principled way to decide for a particular constraint whether it should be blocking or not.